### PR TITLE
feat(sqle): add sqle package

### PR DESCRIPTION
**Pull request by christianmemije**
Test steps

- `npm run build:lib`
- `npm pack ./deploy/ui-platform/`
- Go into another repo 😉  and update the package json to `@td-vantage/ui-platform": "file:../vantage-ui-platform/td-vantage-ui-platform-1.0.0-beta.0.tgz", `(Change path if necessary)
- `npm i` within this new repo
- Pull in module and verify services work as expected

### CODE REVIEW

File: src/lib/sqle/dictionary/dictionary.service.ts [hunk 1]
```diff
@@ -0,0 +1,829 @@
...
+    connection: ISQLEConnection,
+    database: string,
+    table: string,
+    column: string,
+  ): Promise<any> {
+    const requests: any[] = [];
+    const session: any = await this._queryService.createSession(connection).toPromise();
+    const payload: IQueryPayload = {
+      query: `
+        call demouser.td_analyze('frequency', 'database=${database};tablename=${table};columns=${column};');
```
Comment (thomaslennan) [inline src/lib/sqle/dictionary/dictionary.service.ts]: I think demouser should not be reference here? Not sure what to substitute.

File: src/lib/sqle/dictionary/dictionary.service.ts [hunk 2]
```diff
@@ -0,0 +1,829 @@
...
+      frequency: rows,
+    };
+  }
+
+  async getValuesAnalysis(connection: ISQLEConnection, database: string, table: string, column: string): Promise<any> {
+    const requests: any[] = [];
+    const session: any = await this._queryService.createSession(connection).toPromise();
+    const payload: IQueryPayload = {
+      query: `
+        call demouser.td_analyze('values', 'database=${database};tablename=${table};columns=${column};');
```
Comment (thomaslennan) [inline src/lib/sqle/dictionary/dictionary.service.ts]: Ditto here. Not sure what should be in it's place, though. Nickname?

File: src/lib/sqle/dictionary/dictionary.service.ts [hunk 3]
```diff
@@ -0,0 +1,829 @@
+/*
+ * Copyright (C) 2016-2017 by Teradata Corporation. All rights reserved.
```
Comment (thomaslennan) [inline src/lib/sqle/dictionary/dictionary.service.ts]: Remove copyright?

File: src/lib/sqle/query/query.service.ts [hunk 4]
```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright (C) 2016-2017 by Teradata Corporation. All rights reserved.
```
Comment (thomaslennan) [inline src/lib/sqle/query/query.service.ts]: Copyright removed?

File: src/lib/sqle/query/query.service.ts [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+import { map, catchError } from 'rxjs/operators';
+
+/*
+ * These interfaces are duplicated within the query service and the system service.
+ * TODO: DRY this up
+ */
+enum SystemType {
+  Teradata = 'TERADATA',
+  Aster = 'ASTER',
+  Presto = 'PRESTO',
```
Comment (thomaslennan) [inline src/lib/sqle/query/query.service.ts]: Do we support presto anymore? And do these services in general pertain to Teradata?

File: src/lib/sqle/connection/connection.service.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2016-2017 by Teradata Corporation. All rights reserved.
```
[the remark this answers is not in the document]
Comment (christianmemije) [reply src/lib/sqle/connection/connection.service.ts]: Removed

File: src/lib/sqle/dictionary/dictionary.service.ts (hunk 3 above)
Comment (christianmemije) [reply src/lib/sqle/dictionary/dictionary.service.ts]: Removed

File: src/lib/sqle/query/query.service.ts (hunk 4 above)
Comment (christianmemije) [reply src/lib/sqle/query/query.service.ts]: Removed

File: src/lib/sqle/dictionary/dictionary.service.ts (hunk 1 above)
Comment (emoralesb05) [reply src/lib/sqle/dictionary/dictionary.service.ts]: We can probably remove these methods, they were supposed to be used on the profiling functions, but they are still not part of vantage.. so they are leftovers of the PoC.

File: src/lib/sqle/dictionary/dictionary.service.ts (hunk 2 above)
Comment (emoralesb05) [reply src/lib/sqle/dictionary/dictionary.service.ts]: Remove also

File: src/lib/sqle/query/query.service.ts (hunk 5 above)
Comment (emoralesb05) [reply src/lib/sqle/query/query.service.ts]: Yeah, we dont support Presto anymore. For the purpose of the `SQLE` package.. lets remove Aster and Presto from the copied interface.

**Pull request merged**